File: python/alg/linked_list.py
from typing import Optional

class Node:
    def __init__(self, value: int):
        self.value = value
        self.next: Optional[Node] = None

class LinkedList:
    def __init__(self, node: Node = None):
        self.head = node

    def append_head(self, node: Node):
        current = self.head
        self.head = node
        self.head.next = current
    
    def append_tail(self, node: Node):
        if self.head is None:
            self.head = node
            return
       
        current = self.head
        while current.next:
            current = current.next
        current.next = node

    def get_position(self, position: int) -> Optional[Node]:
        if position < 1:
            return None
        
        current = self.head
        while current:
            position -= 1
            if position == 0:
                return current
            current = current.next
        
        return None

    def insert(self, node: Node, position: int):
        if position < 1:
            return
        
        current = self.head
        if position == 1:
            self.head = node
            self.head.next = current
            return
        
        while current:
            position -= 1
            if position == 1:
                old_next = current.next
                current.next = node
                node.next = old_next
                return
            current = current.next

File: golang/structures/linked_list.go
package structures

// Node is a part of Linked List
type Node struct {
	Value int
	Next  *Node
}

// NewNode is a Node constructor
func (n *Node) NewNode(value int) *Node {
	return &Node{
		Value: value,
	}
}

// LinkedList is a linked list data structure
type LinkedList struct {
	Head *Node
}

// LinkedList is a linked list constructor
func (ll *LinkedList) NewLinkedList() *LinkedList {
	return &LinkedList{}
}

// AppendHead appends a node to the head of a linked list
// If node is not nil, otherwise do nothing
func (ll *LinkedList) AppendHead(node *Node) {
	if node == nil {
		return
	}

	if ll.Head != nil {
		current := ll.Head
		ll.Head = node
		ll.Head.Next = current
	}

	ll.Head = node
}

func (ll *LinkedList) AppendTail(node *Node) {
	if node == nil {
		return
	}

	if ll.Head == nil {
		ll.Head = node
		return
	}

	current := ll.Head

	for {
		if current.Next == nil {
			break
		}
		current = current.Next
	}

	current.Next = node
}

func (ll *LinkedList) GetPosition(pos uint) *Node {
	if pos == 0 {
		return nil
	}

	current := ll.Head

	for {
		if current == nil {
			return nil
		}
		pos--
		if pos == 0 {
			return current
		}
		current = current.Next
	}
}
